frontend/packages/shared/src/components/GiteaHeader/VersionControlButtons/VersionControlButtons.tsx

```diff
@@ -1,174 +1,32 @@
-    setSyncModalAnchorEl(currentTarget);
-    setModalState({
-      ...initialModalState,
-      header: t('sync_header.controlling_service_status'),
-      isLoading: true,
-    });
-    const { data: repoStatusResult } = await refetchRepoStatus();
-    if (repoStatusResult) {
-      if (!hasLocalChanges(repoStatusResult) && repoStatusResult.aheadBy === 0) {
-        // if user has nothing to commit => show nothing to push message
-        setModalState({
-          ...initialModalState,
-          shouldShowDoneIcon: true,
-          header: t('sync_header.nothing_to_push'),
-        });
-      } else {
-        // if user has changes to share, show write commit message modal
-        setModalState({
-          ...initialModalState,
-          header: t('sync_header.describe_and_validate'),
-          descriptionText: [
-            t('sync_header.describe_and_validate_sub_message'),
-            t('sync_header.describe_and_validate_sub_sub_message'),
-          ],
-          btnText: t('sync_header.describe_and_validate_btnText'),
-          shouldShowCommitBox: true,
-          btnMethod: commitAndPushChanges,
-        });
-      }
-    }
-  };
-
-  const commitAndPushChanges = async (commitMessage: string) => {
-    setModalState({
-      ...initialModalState,
-      header: t('sync_header.sharing_changes'),
-      isLoading: true,
-    });
-    try {
-      await repoCommitAndPushMutation({ commitMessage });
-    } catch (error) {
-      console.error(error);
-      const { data: result } = await fetchPullData();
-      if (result.hasMergeConflict || result.repositoryStatus === 'CheckoutConflict') {
-        // if pull resulted in a merge conflict, show merge conflict message
-        forceRepoStatusCheck();
-        handleSyncModalClose();
-        setHasMergeConflict(true);
-      }
-      return;
-    }
-
-    const { data: result } = await fetchPullData();
-    if (result.repositoryStatus === 'Ok') {
-      setModalState(initialModalState);
-      setSyncModalAnchorEl(null);
-      toast.success(t('sync_header.sharing_changes_completed'));
-    }
-  };
-
-  const forceRepoStatusCheck = () =>
-    window.postMessage('forceRepoStatusCheck', window.location.href);
+  const { data: repoStatus } = useRepoStatusQuery(org, app);
 
   return (
-    <div className={classes.headerStyling}>
-      <FetchChangesButton
-        fetchChanges={fetchChanges}
-        hasMergeConflict={hasMergeConflict}
-        displayNotification={repoStatus?.behindBy > 0 ?? false}
-        numChanges={repoStatus?.behindBy ?? 0}
-      />
-      <ShareChangesButton
-        hasMergeConflict={hasMergeConflict}
-        hasPushRight={hasPushRights}
-        shareChanges={shareChanges}
-        displayNotification={repoStatus?.contentStatus?.length > 0 ?? false}
-      />
-      <SyncModal anchorEl={syncModalAnchorEl} handleClose={handleSyncModalClose} {...modalState} />
-    </div>
+    <VersionControlButtonsContextProvider
```

Comment:
Great job making this component simpler than it was. 👏 

Do we still need this Context? It seems like we only need to pass onPullSuccess to FetchChangesPopover and ShareChangesPopover. Could the other queries be used directly within each component instead of being added to the context? Avoiding the context might make it easier to handle loading and errors more efficiently, since we then can use the mutations without `mutateAsync` and instead takes advadges from the tanstack query API.

Reply:
That might be a possibility yes, but the `pushRights`, `mergeConflict`, `repoStatus`, `commitAndPushChanges` are also used on different components 🤔 
We can do a call on this too.